canonical-web-and-design/maas-ui
# Remove all deprecationNotices components/actions/selectors/reducers.

## Done
- Removed all deprecationNotices components/actions/selectors/reducers as deprecations now just use the standard notifications api.
- Improved NotificationGroup styling.
- Added dismissable/non-dismissable functionality. If a notification is not dismissable the "Dismiss" link is not shown. If it's part of a group, the "Dismiss all" link will still show but it won't run for that notification.

## QA
- You will need to set up a 2.8 MAAS that was initialised in `all` mode, i.e. the database has been set up inside the snap. This mode has been deprecated in 2.8, so you need to set up a 2.7 MAAS first.
  - Install 2.7 MAAS: `sudo snap install maas --channel=2.7/stable`
  - Initialise in all mode: `sudo maas init` then choose mode `all` and set up a first user
  - Update to 2.8: `sudo snap refresh maas --channel=2.8/edge`
- Checkout maas-ui on master and point the UI at the new MAAS (change the MAAS_URL in `/proxy/env.local`).
- Go through the intro steps until you get to the machine list.
- confirm that there are two deprecation notices, one dismissable and one not.
- Checkout this branch.
- Check that there is only one non-dismissable warning notification about the db being set up inside the snap. You'll probably have another warning about DHCP too, so you will need to expand the warning list.
- If you do have more than one warning, click "Dismiss all" and check that the db deprecation notification remains.
- Run the db migration script: `sudo bash -c '/snap/maas/current/helpers/migrate-snap-database'`
- Wait a minute for MAAS to restart and refresh the machine list. If it doesn't restart automatically, run `sudo snap restart maas`
- Check that the db deprecation warning no longer shows 

## Fixes
Fixes #851 
Fixes #1083 

## Screenshots
### Before
![bolla internal_5240_MAAS_r_machines (1)](https://user-images.githubusercontent.com/25733845/81139873-b27d7480-8faa-11ea-9a79-481a9e4dabe8.png)

### After
![0 0 0 0_8400_MAAS_r_machines (1)](https://user-images.githubusercontent.com/25733845/81139875-b5786500-8faa-11ea-982e-180ef11af437.png)
